.github/workflows/ci.yml

```diff
@@ -0,0 +1,42 @@
+name: Continuous integration
+on: [push]
+jobs:
+  ci:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        go: [ 1.16.x ]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go }}
+      - name: Install dependencies
+        run: |
+          if [ "$RUNNER_OS" == "Linux" ]; then
+            sudo apt update && sudo apt install build-essential
+          fi
+        shell: bash
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: go-${{ matrix.os }}-${{ matrix.go }}-${{ hashFiles('**/go.sum') }}
+      - name: Build
+        run: make all
+      - name: Lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: latest
+          args: "--build-tags=mdbx --config=./.golangci/step1.yml"
+          only-new-issues: true
+          skip-go-installation: true
+          skip-pkg-cache: true
+          skip-build-cache: true
+      - name: Test
+        run: make test
+      - name: Race
+        run: make race
```

Comment:
Folder /tests using 1 test suite with state in it - this is reason why /tests doesn’t support parallel execution of tests and we run all tests with “-p 1” 

-p 1 probably prevents -race to find problems.